src/main/java/org/reaktivity/nukleus/http2/internal/Http2WriteScheduler.java

```diff
@@ -208,6 +208,44 @@ public boolean settingsAck()
         return true;
     }
 
+    @Override
+    public boolean trailers(long traceId, int streamId, byte flags, ListFW<HttpHeaderFW> headers)
+    {
+        MutableDirectBuffer copy = null;
+        int length = headersLength(headers);        // estimate only
+        int sizeof = 9 + headersLength(headers);    // +9 for HTTP2 framing
+        Http2FrameType type = HEADERS;
+        Http2Stream stream = stream(streamId);
+
+        if (buffered() || !hasNukleusBudget(length))
+        {
+            copy = new UnsafeBuffer(new byte[8192]);
+            connection.factory.blockRW.wrap(copy, 0, copy.capacity());
+            connection.mapHeaders(headers, connection.factory.blockRW);
+            HpackHeaderBlockFW block = connection.factory.blockRW.build();
+            length = block.sizeof();
+            sizeof = 9 + length;
+        }
+
+        stream.endStream = true;
+
+        if (buffered() || !hasNukleusBudget(length))
+        {
+            Entry entry = new HeadersEntry(null, streamId, traceId, length, type, flags, copy, 0, length);
+            addEntry(entry);
```

Comment:
In this case, connection.closeStream() would be missing. HeadersEntry doesn't do connection.closeStream(). May be you can create a TrailersEntry (similar to DataEosEntry) which closes the stream.